### PR TITLE
Increased timeout to make the test less fragile

### LINKED ISSUE #8645: LockAdvancedTest Failures
```
java.lang.AssertionError: tryLock() is not interrupted!
    at org.junit.Assert.fail(Assert.java:88)
    at org.junit.Assert.assertTrue(Assert.java:41)
    at com.hazelcast.concurrent.lock.LockAdvancedTest.testLockInterruption(LockAdvancedTest.java:226)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-OracleJDK8/com.hazelcast$hazelcast/886/testReport/junit/com.hazelcast.concurrent.lock/LockAdvancedTest/testLockInterruption/

### PR DESCRIPTION
**Pull request by jerrinot**
Fixes #8645

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/concurrent/lock/LockAdvancedTest.java [hunk 1]
```diff
@@ -388,7 +388,7 @@ public void run() {
         t.start();
         sleepMillis(5000);
         t.interrupt();
-        assertTrue(latch.await(15, TimeUnit.SECONDS));
+        assertTrue(latch.await(30, TimeUnit.SECONDS));
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/concurrent/lock/LockAdvancedTest.java]: Why don't we use assertOpensEventually?

Comment (jerrinot) [inline hazelcast/src/test/java/com/hazelcast/concurrent/lock/LockAdvancedTest.java]: I guess we could, that would provide a nicer failure message. let me change it!

File: hazelcast/src/test/java/com/hazelcast/concurrent/lock/LockAdvancedTest.java [hunk 2]
```diff
@@ -388,7 +388,7 @@ public void run() {
         t.start();
         sleepMillis(5000);
         t.interrupt();
-        assertTrue(latch.await(15, TimeUnit.SECONDS));
+        assertOpenEventually(latch, 30);
```
Comment (jerrinot) [inline hazelcast/src/test/java/com/hazelcast/concurrent/lock/LockAdvancedTest.java]: this is the only change - timeout is increased. everything else is just a conversion of assertions.

**Comment by pveentjer** [comment]
:+1:

**Pull request merged**